+nosnoc/+activeset/Heaviside.m

```diff
@@ -38,7 +38,45 @@
         end
 
         function n_steps = get_n_steps(obj)
-            n_steps = length(obj.regions);
+            n_steps = length(obj.I_0);
+        end
+    end
+
+    methods(Static)
+        function obj = from_pss(pss, model)
+        % Construct a heaviside active set from a pss active set.
+        % In order to calculate we need the S from the model.
+        %
+        % Note:
+        %      For don't cares the realization of the PSS active set is non-unique.
+        %      This means that we make the choice of setting this to the sliding mode, i.e. $\lambda_n = \lambda_p = 0$.
+            times = pss.times;
+            stages = pss.stages;
+            S = model.S{1};
+            I_0 = {};
+            I_1 = {};
+            I_free = {};
+            for ii=1:pss.get_n_steps
+                region_ii = pss.regions{ii};
+                Sp = S(region_ii, :) == 1;
```

Comment:
please add few one line comments in every code chuck. not immediately obvious what is done here.